lightningd/invoice.c

```diff
@@ -281,8 +283,18 @@ static void json_invoice(struct command *cmd,
 	struct preimage r;
 	struct sha256 rhash;
 
-	/* Generate random secret preimage and hash. */
-	randombytes_buf(r.r, sizeof(r.r));
+	if (preimagetok) {
+		/* Get secret preimage from user. */
+		if (!hex_decode(buffer + preimagetok->start,
```

Comment:
If I'm not mistaken this does not raise an error when we provide less than 64 digits.

Reply:
It does: https://github.com/ElementsProject/lightning/blob/9c3691340fa430ed1daf08dc40f76bd7369f66cc/ccan/ccan/str/hex/hex.c#L24-L40

Specifically: https://github.com/ElementsProject/lightning/blob/9c3691340fa430ed1daf08dc40f76bd7369f66cc/ccan/ccan/str/hex/hex.c#L39

If `bufsize == 0` is untrue, or `slen` is odd, then the condition at `return` will fail. At least, my understanding of the code.